app/v2/controllers/V2DeparturesController.scala

```diff
@@ -87,19 +105,46 @@ class V2DeparturesControllerImpl @Inject() (
         withTemporaryFile {
           (temporaryFile, source) =>
             implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
+            val messageType: MessageType   = MessageType.DepartureDeclaration
+            val fileSource                 = FileIO.fromPath(temporaryFile)
+
             (for {
-              result <- validationService.validateJson(MessageType.DepartureDeclaration, source).asPresentation
-              fileSource = FileIO.fromPath(temporaryFile)
-              _          = auditService.audit(AuditType.DeclarationData, fileSource, MimeTypes.JSON)
-              asXml <- conversionService.jsonToXml(MessageType.DepartureDeclaration, fileSource).asPresentation
-              _ = asXml.runWith(Sink.ignore)
-            } yield result).fold[Result](
-              presentationError => Status(presentationError.code.statusCode)(Json.toJson(presentationError)),
-              _ => Accepted
+              _ <- validationService.validateJson(messageType, source).asPresentation
+              _ = auditService.audit(AuditType.DeclarationData, fileSource, MimeTypes.JSON)
+
+              xmlSource         <- conversionService.jsonToXml(messageType, fileSource).asPresentation
+              _                 <- validationService.validateXml(messageType, fileSource).asPresentation
+              declarationResult <- persistAndSendToEIS(xmlSource)
+              _ = xmlSource.runWith(Sink.ignore)
+            } yield declarationResult).fold[Result](
+              presentationError => {
+                fCounter.inc()
+                Status(presentationError.code.statusCode)(Json.toJson(presentationError))
+              },
+              result => {
+                sCounter.inc()
+                Accepted(HateoasDepartureDeclarationResponse(result.departureId))
+              }
             )
         }.toResult
     }
 
+  def persistAndSendToEIS(
+    src: Source[ByteString, _]
+  )(implicit hc: HeaderCarrier, request: AuthenticatedRequest[Source[ByteString, _]]): EitherT[Future, PresentationError, DeclarationResponse] =
+    withTemporaryFileA(
+      src,
+      (temporaryFile, _) => {
+        val fileSource = FileIO.fromPath(temporaryFile)
+        (for {
+          declarationResult <- departuresService.saveDeclaration(request.eoriNumber, fileSource).asPresentation
```

Comment:
Is there scope to deduplicate this with the equivalent section in `submitDeclarationXML`?

Reply:
Done